cmd/load-runner/cli_test.go

```diff
@@ -0,0 +1,173 @@
+package main
+
+import (
+	"flag"
+	"os"
+	"testing"
+
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	"github.com/stretchr/testify/assert"
+
+	"rudder-load/internal/parser"
+)
+
+func TestCLI_ParseFlags(t *testing.T) {
+	// Save original args and restore after test
+	originalArgs := os.Args
+	defer func() { os.Args = originalArgs }()
+
+	tests := []struct {
+		name        string
+		args        []string
+		want        *parser.CLIArgs
+		wantErr     bool
+		errContains string
+	}{
+		{
+			name: "valid direct args",
+			args: []string{
+				"cmd",
+				"-d", "1h",
+				"-n", "test-namespace",
+				"-l", "test-load",
+				"-f", "path/to/chart",
+			},
+			want: &parser.CLIArgs{
+				Duration:       "1h",
+				Namespace:      "test-namespace",
+				LoadName:       "test-load",
+				ChartFilesPath: "path/to/chart",
+				TestFile:       "",
+			},
+			wantErr: false,
+		},
+		{
+			name: "valid test file arg",
+			args: []string{
+				"cmd",
+				"-t", "tests/test.yaml",
+			},
+			want: &parser.CLIArgs{
+				TestFile: "tests/test.yaml",
+			},
+			wantErr: false,
+		},
+		{
+			name: "missing required args without test file",
+			args: []string{
+				"cmd",
+				"-n", "test-namespace",
+				"-l", "test-load",
+			},
+			want:        nil,
+			wantErr:     true,
+			errContains: "invalid options",
+		},
+		{
+			name: "missing namespace",
+			args: []string{
+				"cmd",
+				"-d", "1h",
+				"-l", "test-load",
+			},
+			want:        nil,
+			wantErr:     true,
+			errContains: "invalid options",
+		},
+		{
+			name: "missing load name",
+			args: []string{
+				"cmd",
+				"-d", "1h",
+				"-n", "test-namespace",
+			},
+			want:        nil,
+			wantErr:     true,
+			errContains: "invalid options",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Reset flags before each test
+			flag.CommandLine = flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+			// Set up test args
+			os.Args = tt.args
+
+			cli := NewCLI(logger.NewLogger())
```

Comment:
You might want to use `logger.NOP` for tests.

Reply:
Got it. done.